tests/l2/qat/qatlib_rbac.yaml

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) 2022 - 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: intel-qat
+  namespace: intel-qat
+---
+apiVersion: security.openshift.io/v1
+allowHostDirVolumePlugin: false
+allowHostIPC: false
+allowHostNetwork: false
+allowHostPID: false
+allowHostPorts: false
+allowPrivilegeEscalation: false
+allowPrivilegedContainer: false
+allowedCapabilities: 
+- IPC_LOCK
+defaultAddCapabilities: null
+fsGroup:
+   type: MustRunAs
+groups: []
+kind: SecurityContextConstraints
+metadata:
+  annotations:
+    kubernetes.io/description: 'SCC to use IPC_LOCK capability for qatlib pod'
+  name: container-scc
```

Comment:
Looks like we are creating a new SCC for qat workload using qatlib. We should be very carefully to create and maintain and use a user defined SCC see https://docs.openshift.com/container-platform/4.13/authentication/managing-security-context-constraints.html#security-context-constraints-creating_configuring-internal-oauth.
If we can't use the predefined SCC by OCP https://docs.openshift.com/container-platform/4.13/authentication/managing-security-context-constraints.html#default-sccs_configuring-internal-oauth
and have to create our own SCC. I suggest starting from predefined "restricted" or "restricted-v2" SCC and only adding the "IPC_LOCK" capability. To make the user defined SCC more easily be maintained, I suggest to have a single yaml file to maintain it

Reply:
> Looks like we are creating a user-defined SCC for qat workload using qatlib.

It's for all containers using QAT through VFIO (also DPDK's libs and DPDK test apps like `crypto/compress-perf` belong to this)

Reply:
> Looks like we are creating a user-defined SCC for qat workload using qatlib. We should be very careful to create,  I suggest starting from predefined "restricted" or "restricted-v2" SCC and only adding the "IPC_LOCK" capability. To make the user-defined SCC more easily maintained, I suggest having a single yaml file to maintain it
The pod needs to run as root, via restricted it wont work. Default capability with IPC_LOCK defaults to anyuid . The SCC here does not give access to file system and only executes as root with IPC_LOCK. Also, if other workloads need any other permissions, we just need to update this yaml to add it and in the job.

Reply:
#35 is about making it possible to run containers as non-root if it helps to avoid custom SCCs

Reply:
Thanks, tried this. For qatlib, it has a prereq to run as root https://github.com/intel/qatlib/blob/7429ee2b7c837137ed11959a3c2cc3729dc15739/INSTALL#L60. Will be following for gpu and sgx in 1.1.0 release

Reply:
> For qatlib, it has a prereq to run as root

@vbedida79 it's not true. At least in our orchestration case we can run QAT as non-root just like with gpu/sgx

Reply:
if you wish to test it non-root, CRI-O must be configured with:

```
[crio.runtime]
device_ownership_from_security_context = true
```

Reply:
got it, on the host node then. this might be it. what do you think @mregmi and @uMartinXu